apps/lend/src/entities/chain/lib/data-hooks.ts

```diff
@@ -0,0 +1,26 @@
+import { useOneWayMarketNames } from './query-hooks'
+import networks from '@/networks'
+import { OneWayMarketTemplate } from '@curvefi/lending-api/lib/markets'
+import { useMemo } from 'react'
+import { useApi } from './chain-info'
+import { checkChainValidity } from '@/entities/chain'
+
+export const useOneWayMarketMapping = (chainId: ChainId) => {
+  const chainValid = checkChainValidity({ chainId }) // extra check to make sure the API is loaded before we use stale market names
+
+  const { data: marketNames, ...rest } = useOneWayMarketNames({ chainId })
+  const { data: api } = useApi();
+  const data: Record<string, OneWayMarketTemplate> | undefined = useMemo(() =>
+    chainValid && marketNames && api ? Object.fromEntries(
+      marketNames
+        .filter(marketName => !networks[chainId].hideMarketsInUI[marketName])
+        .map(name => [name, api.getOneWayMarket(name)])
+    ) : undefined, [api, chainId, chainValid, marketNames]
+  )
+  return { data, ...rest }
+}
+
+export const useOneWayMarket = (chainId: ChainId, marketName: string) => {
```

Comment:
Good to add type to the marketName

Reply:
what do you mean? it shouldn't be `: string`?

Reply:
I assumed that `getMarketList` returns a special type, not just a string. This does not seem to be the case.